lib/cloud-metadata/index.js

```diff
@@ -0,0 +1,197 @@
+'use strict'
+const { getMetadataAwsV1, getMetadataAwsV2 } = require('./aws')
+const { getMetadataAzure } = require('./azure')
+const { getMetadataGcp } = require('./gcp')
+const { CallbackCoordination } = require('./callback-coordination')
+const DEFAULT_PROVIDER_CONFIG = {
+  aws: {
+    host: '169.254.169.254',
+    port: 80,
+    protocol: 'http'
+  },
+  azure: {
+    host: '169.254.169.254',
+    port: 80,
+    protocol: 'http'
+  },
+  gcp: {
+    host: 'metadata.google.internal',
+    port: 80,
+    protocol: 'http'
+  }
+}
```

Comment:
Let's talk monday about the motivation for moving them into the individual module -- for that it's worth they ended up in this file because I wanted to have the values injectable via the `getCloudMetadata` server.  That gets more awkward (or maybe just moves the awkwardness) if the defaults are in each individual module.

Reply:
I moved the default URL values to their respective modules, turned them into `URL` objects, and modified the methods signature's to accepts a single `baseUrlOverride` parameter for setting a different value.  It was a persnickety refactor, but the API's simpler now and that's always a 👍